content/docs/how-we-work/coding-principles-standards.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Coding Principles and Standards
+weight: 4
+description: At Axelerant, we write code not only to solve a specific problem but also to provide a reusable solution that can be used in a community level. We have principles that can guide us to the desired code standards.
+---
+
+# Coding Principles and Standards
+
+At Axelerant, we write code not only to solve a specific problem but also to provide a reusable solution that can be used in a community level. We have principles that can guide us to the desired code standards.
+
+## Coding Principles
+
+- We use the correct layer of abstraction.
+
+- We write clean, simple code and explain our purpose with comments.
+
+- We contribute back everything we can, because it supports us, our clients and the community.
+
+- We manage patches in a standard way so that we can know if code/contrib has been changed, how and why. Every change should be contributed back, and the patch name should include the drupal.org issue NID.
+
+- We write secure code.
+
+- We write code that anticipates change, so that we can implement (most) changes in an easy and Agile way.
+
+- We write API layer that doesn't assume a particular UI, so that the API could be used in multiple (and unpredictable) situations (For example, used in these UIs: Web site, drush, rest, SMS).
+
+- We remove dead code, because it's dead (and because it's confusing to leave it).
+
+- We treat one as a special case of many, so that the logic is in one place and we don't have to maintain it in two places.
+
+- When writing getter functions, consider handling the logic in the many only. A `get_one` can call the `get_many`.
+
+- We use doxygen commenting standards on the top of functions.
+
+- We include implementation notes and end-user documentation in our tickets.
+
+## Coding Standards
+
+- We follow established best practices so we can easily work with each others code, and contribute back.
+
+- We don't hard code
+
+- We stand behind the YAGNI (You ain't gonna need it) philosophy and avoid "gold plating"
+
+  - When writing code, our code should solve ONLY and EXACTLY the use case that we are trying to support. In many instances we have a desired to make things MORE FLEXIBLE for the FUTURE but that is code we will have to support without knowing whether any of those use cases will ever come up. Another big reason for wanting to be FLEXIBLE is that refactoring is hard, and it feels easier to have flexibility at front, but that is why we write automated tests, so refactoring will not be as painful.
+
+### Drupal-specific Coding Standards
+
+We follow [Drupal Coding Standards](https://www.drupal.org/docs/develop/standards) which includes PHP, CSS, JavaScript, Twig, YAML, and more.
```

Comment:
Should this have been a bullet point as well?

Reply:
They were in a bullet but the link there can take us to each items mentioned.